pkg/unixserver/server.go

```diff
@@ -12,6 +12,9 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+//MAXTTL for removing metrics after <value> seconds of inactivity
+var MAXTTL float64 = 300
```

Comment:
I know you already flagged this, but just to copy the suggestion from our Trello board here where it can be seen:

My first instinct is to follow the example of stale metrics in prometheus and suggest something like expiring after 5x the interval. OTOH, 5s (for a 1/s metric) seems like a very short TTL somehow. Maybe we should try: `min(default(configurable_timeout, 300), 5 * metric_interval)`

IIRC the raw messages coming from collectd have interval information in them, though I'm not seeing it make it all the way to prometheus currently, and I haven't looked at why (or whether my memory is just mistaken)

Reply:
Oh they do? That makes this much easier then. Even if Prometheus doesn't store this information for whatever reason, it can be added to the metric object and time out intervals can be generated from that

Reply:
I don't think Prometheus has a mechanism to transfer the collection interval since I Prometheus assumes it is driving the interval.   I would vote for the 5x interval or smaller.  Don't propagate the lie longer than needed :).